synapse/module_api/__init__.py

```diff
@@ -1666,6 +1667,46 @@ async def set_displayname(
             deactivation=deactivation,
         )
 
+    R = TypeVar("R")
+
+    def run_as_background_process(
+        self,
+        desc: str,
+        func: Callable[..., Awaitable[Optional[R]]],
+        *args: Any,
+        bg_start_span: bool = True,
+        **kwargs: Any,
+    ) -> "defer.Deferred[Optional[R]]":
+        """Run the given function in its own logcontext, with resource metrics
+
+        This should be used to wrap processes which are fired off to run in the
+        background, instead of being associated with a particular request.
+
+        It returns a Deferred which completes when the function completes, but it doesn't
+        follow the synapse logcontext rules, which makes it appropriate for passing to
+        clock.looping_call and friends (or for firing-and-forgetting in the middle of a
+        normal synapse async function).
```

Comment:
This description seems quite entangled to Synapse internals, here is a suggestion trying to be a bit more generic.
```suggestion
        """Run the given function in the background, with its own logcontext.

        This should be used to wrap processes which are fired off to run in the
        background.

        It returns a Deferred which completes when the function completes, but it doesn't
        follow the synapse logcontext rules.
```

Reply:
I've went with Sean's suggestion to below which eliminates the need for any of this, but thank you for pointing it out! It's helpful to know how to think about these things.